stellar/txobserver/main.go

```diff
@@ -21,60 +23,55 @@ type Observer struct {
 	store  *store.DB
 	log    *slog.Entry
 
-	ledgerSequence uint32
+	bridgeAccountBirthSequence uint32
 }
 
-func NewObserver(ctx context.Context, bridgeAccount string, client *horizonclient.Client, store *store.DB) *Observer {
+func NewObserver(
+	ctx context.Context,
+	bridgeAccount string,
+	bridgeAccountBirthSequence uint32,
+	client *horizonclient.Client,
+	store *store.DB,
+) *Observer {
 	o := &Observer{
-		ctx:           ctx,
-		bridgeAccount: bridgeAccount,
-		client:        client,
-		store:         store,
-		log:           slog.DefaultLogger.WithField("service", "stellar_txobserver"),
-	}
-
-	ledgerSeq, err := o.store.GetLastLedgerSequence(context.Background())
-	if err != nil {
-		o.log.Fatalf("Unable to load last ledger sequence from db: %v", err)
-	}
-
-	if ledgerSeq == 0 {
-		root, err := o.client.Root()
-		if err != nil {
-			o.log.Fatalf("Unable to access Horizon (%s) root resource: %v", client.HorizonURL, err)
-		}
-
-		o.ledgerSequence = uint32(root.HorizonSequence)
-	} else {
-		o.ledgerSequence = ledgerSeq
+		ctx:                        ctx,
+		bridgeAccount:              bridgeAccount,
+		bridgeAccountBirthSequence: bridgeAccountBirthSequence,
+		client:                     client,
+		store:                      store,
+		log:                        slog.DefaultLogger.WithField("service", "stellar_txobserver"),
 	}
 
 	return o
 }
 
 func (o *Observer) ProcessNewLedgers() {
 	for o.ctx.Err() == nil {
-		// Get ledger data first to ensure there are no gaps
-		ledger, err := o.client.LedgerDetail(o.ledgerSequence)
-		if err != nil {
-			if herr, ok := err.(*horizonclient.Error); ok && herr.Response.StatusCode == http.StatusNotFound {
-				// Ledger not found means we reached the latest ledger
-				return
+		if ledgerSeq, err := o.store.GetLastLedgerSequence(context.Background()); err != nil {
+			o.log.Errorf("Unable to load last ledger sequence from db: %v", err)
+		} else {
+			if ledgerSeq == 0 {
+				ledgerSeq = o.bridgeAccountBirthSequence
```

Comment:
There is a risk of skipping some transactions if Starbridge is connected to Horizon with a partial history. I think I solved it in https://github.com/stellar/starbridge/pull/70. Let's say it's not a blocker here and we can discuss it later in https://github.com/stellar/starbridge/pull/70.